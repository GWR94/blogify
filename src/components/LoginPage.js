import React from 'react';
import { connect } from 'react-redux';
import {
	startLoginWithFacebook,
	startLoginWithGoogle,
	startLoginWithGithub,
	startLoginWithTwitter,
} from '../actions/auth';
import { Link } from 'react-router-dom';
import { startGetAllPosts } from '../actions/posts';

export class LoginPage extends React.Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<div className="box-layout">
				<div className="box-layout__box">
					<h1 className="box-layout__title">Boilerplate</h1>
					<p>Tag line for app</p>
					<button onClick={this.props.startLoginWithGoogle} className="button__login button__login--google">
						<i className="fa fa-google" aria-hidden="true" />Login With Google
					</button>
					<button
						onClick={this.props.startLoginWithFacebook}
						className="button__login button__login--facebook"
					>
						<i className="fa fa-facebook" aria-hidden="true" />Login With Facebook
					</button>
					<button onClick={this.props.startLoginWithTwitter} className="button__login button__login--twitter">
						<i className="fa fa-twitter" aria-hidden="true" />Login With Twitter
					</button>
					<button onClick={this.props.startLoginWithGithub} className="button__login button__login--github">
						<i className="fa fa-github" aria-hidden="true" />Login With GitHub
					</button>
					<Link to="/dashboard" className="button button__login" onClick={this.props.startGetAllPosts}>
						<i className="fa fa-search" />Search
					</Link>
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = dispatch => ({
	startLoginWithFacebook: () => dispatch(startLoginWithFacebook()),
	startLoginWithGoogle: () => dispatch(startLoginWithGoogle()),
	startLoginWithGithub: () => dispatch(startLoginWithGithub()),
	startLoginWithTwitter: () => dispatch(startLoginWithTwitter()),
	startGetAllPosts: () => dispatch(startGetAllPosts()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
