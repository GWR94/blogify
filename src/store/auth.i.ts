import { Post, S3Image } from "./posts.i";

export const LOGIN = "LOGIN";
export const LOGOUT = "LOGOUT";
export const UPDATE_USER = "UPDATE_USER";

export interface AuthState {
  uid: string | null;
  name: string | null;
  email: string | null;
  following: string[];
  followers: string[];
}

export interface User {
  id?: string;
  username: string;
  email: string;
  name: string;
  following: string[];
  followers: string[];
  savedPosts: string[];
  posts?: {
    items: Post[];
  };
  profileImage: S3Image | null;
}

export interface LoginAction {
  uid: string;
  name: string;
  email: string;
}

export interface UpdateUserAction {
  updates: Partial<AuthState>;
}

export interface LogoutAction {
  type: typeof LOGOUT;
}

export declare type AuthActionTypes = LoginAction | LogoutAction | UpdateUserAction;
