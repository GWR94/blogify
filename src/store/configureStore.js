import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import authReducer from '../reducers/auth';
import postsReducer from '../reducers/posts';
import filtersReducer from '../reducers/filters';
import searchReducer from '../reducers/search';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default () => {
	const store = createStore(
		combineReducers({
			auth: authReducer,
			posts: postsReducer,
			filters: filtersReducer,
			search: searchReducer
		}),
		composeEnhancers(applyMiddleware(thunk))
	);

	return store;
};
