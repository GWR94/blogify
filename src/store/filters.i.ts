import { Moment } from "moment";

export interface FiltersState {
  text: string;
  sortBy: "title" | "tags" | "title" | "author";
  startDate: Moment;
  endDate: Moment;
}

export const SET_TEXT_FILTER = "SET_TEXT_FILTER";
export const SORT_BY_TAGS = "SORT_BY_TAGS";
export const SORT_BY_TITLE = "SORT_BY_TITLE";
export const SORT_BY_AUTHOR = "SORT_BY_AUTHOR";
export const SET_START_DATE = "SET_START_DATE";
export const SET_END_DATE = "SET_END_DATE";

export interface SetTextFilterAction {
  type: typeof SET_TEXT_FILTER;
  text: string;
}

export interface SortByTagsAction {
  type: typeof SORT_BY_TAGS;
}

export interface SortByAuthorAction {
  type: typeof SORT_BY_AUTHOR;
}

export interface SortByTitleAction {
  type: typeof SORT_BY_TITLE;
}

export interface SetStartDateAction {
  type: typeof SET_START_DATE;
  startDate: Moment;
}

export interface SetEndDateAction {
  type: typeof SET_END_DATE;
  endDate: Moment;
}

export declare type FiltersActionTypes =
  | SetEndDateAction
  | SetStartDateAction
  | SetTextFilterAction
  | SortByAuthorAction
  | SortByTagsAction
  | SortByTitleAction;
