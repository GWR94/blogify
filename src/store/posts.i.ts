export const GET_POST = "GET_POST";
export const ADD_POST = "ADD_POST";
export const REMOVE_POST = "REMOVE_POST";
export const EDIT_POST = "EDIT_POST";
export const SET_POSTS = "SET_POSTS";
export const SET_PUBLIC_POSTS = "SET_PUBLIC_POSTS";
export const SET_COMMENTS = "SET_COMMENTS";
export const UPDATE_COMMENT = "UPDATE_COMMENTS";
export const DELETE_COMMENT = "DELETE_COMMENT";
export const ADD_COMMENT = "ADD_COMMENT";
export const LOAD_MORE_POSTS = "LOAD_MORE_POSTS";

export interface PostsState {
  posts: Post[];
  nextToken: string | null;
  comments: Comment[];
}

export interface Comment {
  id: string;
  content: string;
  likes: string[];
  postID: string;
  userID: string;
}

export interface Post {
  author: string;
  body: string;
  email: string;
  id?: string;
  createdAt?: string;
  overview: string;
  tags: string[];
  title: string;
  userID?: string;
  public: boolean;
}

export interface S3Image {
  key: string;
  bucket: string;
  region: string;
}

export interface UpdateCommentAction {
  id: string;
  updates: Partial<Comment>;
}

export interface AddPostAction {
  post: Post;
}

export interface SetCommentsAction {
  comments: Comment[];
}

export interface DeleteCommentAction {
  id: string;
}

export interface AddCommentAction {
  comment: Comment;
}

export interface RemovePostAction {
  id: string;
}

export interface EditPostAction {
  id: string;
  updates: Partial<Post>;
}

export interface SetPostsAction {
  posts: Post[];
  nextToken: string | null;
}

export interface SetPublicPostsAction {
  posts: Post[];
  nextToken: string | null;
}

export interface LoadMorePostsAction {
  posts: Post[];
  nextToken: string | null;
}

export declare type PostsActionTypes =
  | SetPostsAction
  | SetPublicPostsAction
  | EditPostAction
  | RemovePostAction
  | AddPostAction
  | AddCommentAction
  | UpdateCommentAction
  | DeleteCommentAction
  | SetCommentsAction
  | LoadMorePostsAction;
