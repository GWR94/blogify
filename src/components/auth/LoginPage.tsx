import React from "react";
import { Typography } from "@material-ui/core";
import { Auth } from "aws-amplify";
import { useSelector } from "react-redux";
import { useHistory } from "react-router-dom";
import Button from "../../utils/components/MuiButton";
import { openSnackbar } from "../../utils/components/Notifier";
import { AppState } from "../../store/store";
import background from "./img/bg.jpg";

export enum CognitoHostedUIIdentityProvider {
  Cognito = "COGNITO",
  Google = "Google",
  Facebook = "Facebook",
  Amazon = "LoginWithAmazon",
  Apple = "SignInWithApple",
}

const LoginPage = (): JSX.Element => {
  const { uid } = useSelector(({ auth }: AppState) => auth);
  const history = useHistory();
  if (!uid) history.push("/dashboard");
  return (
    <div className="box-layout" style={{ background: `url(${background})` }}>
      <div className="box-layout__box">
        <div>
          <Typography
            variant="h3"
            className="box-layout__title"
            style={{
              fontSize: "30px",
              marginBottom: "14px",
              fontWeight: 600,
            }}
          >
            Blogify
          </Typography>
          <Typography gutterBottom>
            Create and share posts that are interesting to{" "}
            <span style={{ fontWeight: "bold" }}>you.</span>
          </Typography>
          <Typography style={{ fontSize: "15px", marginBottom: 18 }}>
            Let the world know what you&apos;re up to.
          </Typography>
          <button
            type="button"
            onClick={async (): Promise<void> => {
              try {
                await Auth.federatedSignIn({
                  provider: CognitoHostedUIIdentityProvider.Google,
                });
                history.push("/dashboard");
                openSnackbar({
                  message: "Successfully logged in.",
                  severity: "success",
                });
              } catch (err) {
                openSnackbar({
                  message: "Unable to login. Please try again.",
                  severity: "error",
                });
              }
            }}
            className="button__login--google"
          >
            <i className="fa fa-google" aria-hidden="true" />
          </button>
          <button
            type="button"
            onClick={async (): Promise<void> => {
              try {
                await Auth.federatedSignIn({
                  provider: CognitoHostedUIIdentityProvider.Facebook,
                });
                history.push("/dashboard");
                openSnackbar({
                  message: "Successfully logged in.",
                  severity: "success",
                });
              } catch (err) {
                openSnackbar({
                  message: "Unable to login. Please try again.",
                  severity: "error",
                });
              }
            }}
            className="button__login--facebook"
          >
            <i className="fa fa-facebook" aria-hidden="true" />
          </button>
          <button
            type="button"
            onClick={async (): Promise<void> => {
              try {
                await Auth.federatedSignIn({
                  provider: CognitoHostedUIIdentityProvider.Amazon,
                });
                history.push("/dashboard");
                openSnackbar({
                  message: "Successfully logged in.",
                  severity: "success",
                });
              } catch (err) {
                openSnackbar({
                  message: "Unable to login. Please try again.",
                  severity: "error",
                });
              }
            }}
            className="button__login--amazon"
          >
            <i className="fa fa-amazon" aria-hidden="true" />
          </button>
          <Typography style={{ fontSize: "17px", margin: "0 20px 18px" }}>
            Login with any social media or search browse through public posts by clicking
            &quot;Search&quot; below
          </Typography>
          <Button
            color="primary"
            className="button"
            variant="contained"
            onClick={(): void => history.push("/dashboard?view=all")}
            style={{ textDecoration: "none", color: "white" }}
          >
            <i className="fa fa-search" style={{ marginRight: 5 }} /> Search
          </Button>
        </div>
      </div>
    </div>
  );
};

export default LoginPage;
