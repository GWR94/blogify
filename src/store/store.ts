import { configureStore, Store, compose } from "@reduxjs/toolkit";
import authReducer from "../reducers/auth.reducer";
import postsReducer from "../reducers/posts.reducer";
import filtersReducer from "../reducers/filters.reducer";
import { AuthState } from "./auth.i";
import { FiltersState } from "./filters.i";
import { PostsState } from "./posts.i";
import { authSlice } from "../slices/auth.slice";
import { postsSlice } from "../slices/post.slice";
import { filtersSlice } from "../slices/filter.slice";

declare global {
  interface Window {
    // eslint-disable-next-line no-undef
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

export interface GraphQLResult<T = Record<string, unknown>> {
  data?: T;
  errors?: [Record<string, unknown>];
  extensions?: {
    [key: string]: any;
  };
}

export interface AppState {
  auth: AuthState;
  posts: PostsState;
  filters: FiltersState;
}

export type RootState = ReturnType<typeof store.getState>;

export const store: Store = configureStore({
  reducer: {
    auth: authSlice.reducer,
    posts: postsSlice.reducer,
    filters: filtersSlice.reducer,
  },
});

export default store;
