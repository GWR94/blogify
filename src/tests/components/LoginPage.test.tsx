import React from "react";
import LoginPage from "../../components/auth/LoginPage";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore, { MockStore } from "redux-mock-store";
import { AppState } from "../../store/store";
import moment from "moment";
/**
 * @jest-environment jsdom
 */

const initialState: AppState = {
  auth: {
    uid: "123",
    name: "Bill",
    email: "bill@test.com",
    followers: [],
    following: [],
  },
  posts: {
    posts: [],
    nextToken: null,
    comments: [],
  },
  filters: {
    text: "",
    sortBy: "title",
    startDate: moment(100),
    endDate: moment(1000),
  },
};

const mockStore = configureStore();

describe("<LoginPage />", () => {
  it("should render correctly when the user is logged in", () => {
    let store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <LoginPage />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
