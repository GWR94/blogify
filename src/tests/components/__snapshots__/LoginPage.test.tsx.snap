// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render correctly when the user is logged in 1`] = `
<DocumentFragment>
  <div
    class="box-layout"
  >
    <div
      class="box-layout__box"
    >
      <div>
        <h3
          class="MuiTypography-root box-layout__title MuiTypography-h3"
          style="font-size: 30px; margin-bottom: 14px; font-weight: 600;"
        >
          Blogify
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Create and share posts that are interesting to 
          <span
            style="font-weight: bold;"
          >
            you.
          </span>
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
          style="font-size: 15px; margin-bottom: 18px;"
        >
          Let the world know what you're up to.
        </p>
        <button
          class="button__login--google"
          type="button"
        >
          <i
            aria-hidden="true"
            class="fa fa-google"
          />
        </button>
        <button
          class="button__login--facebook"
          type="button"
        >
          <i
            aria-hidden="true"
            class="fa fa-facebook"
          />
        </button>
        <button
          class="button__login--amazon"
          type="button"
        >
          <i
            aria-hidden="true"
            class="fa fa-amazon"
          />
        </button>
        <p
          class="MuiTypography-root MuiTypography-body1"
          style="font-size: 17px; margin: 0px 20px 18px;"
        >
          Login with any social media or search browse through public posts by clicking "Search" below
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          style="text-decoration: none; color: white;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <i
              class="fa fa-search"
              style="margin-right: 5px;"
            />
             Search
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
